test/support/query.ex

```diff
@@ -0,0 +1,5 @@
+defmodule Authsense.Test.Query do
+  alias Authsense.Test.Repo
+
+  def where(_model, extra_field: extra_field), do: Enum.filter(Repo.all(), &(&1.extra_field == extra_field))
```

Comment:
Don't use `Enum` to traverse repo transactions. In production apps with 100k users, this will be a nightmare in performance. I know this is just a test, but still...

Also, you're discarding `_model`.

```rb
def where(model, extra_field) do
  query = from u in Model, where: u.extra_field == ^extra_field
  Repo.all query
end
```

Or better yet, just rewrite your tests such that:

```elixir
  test "authenticate with opts and retrieve nothing" do
      query = from u in User, where: u.extra_field == "newbie"

      assert {:error, _error} = %User{}
      |> change(%{email: "rico@gmail.com", password: "foobar"})
      |> Service.authenticate(scope: (fn -> Repo.all(query) end))
  end
```